### PR TITLE
Added support for Multitask regression task in HuggingFace models 

### PR DESCRIPTION
**Pull request by arunppsg**
## Description

I have updated `HuggingFaceModel` class to allow for different type of tasks.

Earlier, task can only be `finetuning` or `pretraining`. If the task was `pretraining`, then the default 
pretraining mode assumed is masked language modeling and data was prepared accordingly. The problem with this approach was other pretraining task like multitask regression uses different data batching process (the batching does not involve masking as required for mlm) and hence, there was no way to generate data for tasks like `mtr`.

The PR broadens the scope of task by including various types of task like `mlm` for masked language modeling, `mtr` for multitask regression, `regression` and `classification` for simple regression and classification tasks. Task of type `finetuning` is removed since finetuning itself involves multiple subtasks like regression, classification, mtr. Instead, users can directly specify the task.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rbharath — APPROVED** [review]
LGTM
